src/platform/common/application/notebook.ts

```diff
@@ -112,10 +113,11 @@ export class VSCodeNotebook implements IVSCodeNotebook {
     public async closeActiveNotebooks() {
         // We could have untitled notebooks, close them by reverting changes.
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const documents = new Set<NotebookDocument>();
+        const documents = new Set<NotebookDocument>(this.notebookDocuments);
```

Comment:
Never mind, i can see that we're relying on the events.. bummer

Reply:
Yes it is a hack. The 'revertAndClose' command does not actually send a close event. It can't be in the test code because it can't fire the onDidCloseNotebook event. Only this class can fire that event.